lib/OpenQA/Setup.pm

```diff
@@ -328,6 +328,60 @@ sub load_plugins {
     OpenQA::Setup::update_config($server->config, @{$server->plugins->namespaces}, "OpenQA::WebAPI::Auth");
 }
 
+sub load_ui_plugins {
+    my ($server, $plugin_route) = @_;
+
+    return if (!defined $server->config->{ini_config} || !defined $server->config->{ini_config}->{v});
+
+    push @{$server->plugins->namespaces}, 'WebAPIPlugin';
+    for my $section (@{$server->config->{ini_config}->{sects}}) {
+        my $plugin;
+        if ($section =~ /WebAPIPlugin(.+)/) {
+            $plugin = $1;
+        }
+        else {
+            next;
+        }
+        my $plugin_settings = $server->config->{ini_config}->{v}->{$section};
+        next if (!$plugin_settings->{enabled});
+        $server->log->info("Loading plugin $section...");
+        my $res = eval {
+            my $path     = $plugin_settings->{path};
+            my $jsonfile = "$path/WebAPIPlugin/$plugin.json";
+            $server->log->info("Checking $jsonfile...");
+            if (-f $jsonfile) {
+                # load routes if defined
+                $server->plugin(
+                    OpenAPI => {
+                        route => $plugin_route,
+                        url   => $jsonfile
+                    });
+                push @{$server->renderer->paths}, $path . '/WebAPIPlugin';
+                $server->log->info("Loaded $jsonfile");
+            }
+            my $res = eval "use lib '$path';\nuse WebAPIPlugin::$plugin;\n1;";
+            if (!defined $res) {
+                die $@;
+            }
```

Comment:
`die $@ unless defined $res;`?

Reply:
Something like this should work.
```perl
use Mojo::Loader 'load_class';
push @INC, $path;
my $e = load_class "WebAPIPlugin::$plugin";
```

Reply:
I actually forgot, the `->plugin()` call below already does a `load_class` for you. So it might be enough to modify `@INC` and you're already done.

Reply:
->plugin() fails here with `Plugin "OpenQA::Plugin::WebAPI::Hello" missing, maybe you need to install it?`, unless I do `eval "use $plugin;"`

Reply:
Why did you tell it to load `OpenQA::Plugin::WebAPI::Hello`? Load the correct class by entering the full name, or update the plugin namespaces and use the relative name.
```perl
unshift @{$app->plugins->namespaces}, 'WebAPIPlugin';
$app->plugin('Hello');
```